app/src/app/components/sidebar/ResetMapButton.tsx

```diff
@@ -2,16 +2,11 @@ import {useMapStore} from '@/app/store/mapStore';
 import {Button} from '@radix-ui/themes';
 
 export function ResetMapButton() {
-  const mapStore = useMapStore.getState();
-
-  const handleClickResetMap = () => {
-    mapStore.setFreshMap(true);
-    // clear map metrics
-    mapStore.setMapMetrics(null);
-  };
+  const handleClickResetMap = useMapStore(state => state.handleReset)
+  const noZonesAreAssigned = useMapStore(state => !state.zoneAssignments.size)
```

Comment:
OOOooohh love how you're just doing the bang in the callback

Reply:
ty !